lib/openai/client.rb

```diff
@@ -52,37 +52,62 @@ def translate(parameters: {})
     end
 
     def self.get(path:)
-      HTTParty.get(
-        uri(path: path),
-        headers: headers,
-        timeout: request_timeout
-      )
+      to_json(conn.get(uri(path: path), timeout: OpenAI.configuration.request_timeout) do |req|
+        req.headers = headers
+      end&.body)
     end
 
     def self.json_post(path:, parameters:)
-      HTTParty.post(
-        uri(path: path),
-        headers: headers,
-        body: parameters&.to_json,
-        timeout: request_timeout
-      )
+      to_json(conn.post(uri(path: path), timeout: OpenAI.configuration.request_timeout) do |req|
+        if parameters[:stream].is_a?(Proc)
+          req.options.on_data = to_json_stream(user_proc: parameters[:stream])
+          parameters[:stream] = true # Necessary to tell OpenAI to stream.
+        end
+
+        req.headers = headers
+        req.body = parameters.to_json
+      end&.body)
     end
 
     def self.multipart_post(path:, parameters: nil)
-      HTTParty.post(
-        uri(path: path),
-        headers: headers.merge({ "Content-Type" => "multipart/form-data" }),
-        body: parameters,
-        timeout: request_timeout
-      )
+      m_conn = Faraday.new { |f| f.request :multipart }
+
+      to_json(m_conn.post(uri(path: path), timeout: OpenAI.configuration.request_timeout) do |req|
+        req.headers = headers.merge({ "Content-Type" => "multipart/form-data" })
+        req.body = multipart_parameters(parameters)
+      end&.body)
     end
 
     def self.delete(path:)
-      HTTParty.delete(
-        uri(path: path),
-        headers: headers,
-        timeout: request_timeout
-      )
+      to_json(conn.delete(uri(path: path), timeout: OpenAI.configuration.request_timeout) do |req|
+        req.headers = headers
+      end&.body)
+    end
+
+    def self.to_json(string)
+      return unless string
+
+      JSON.parse(string)
+    rescue JSON::ParserError
+      # Convert a multiline string of JSON objects to a JSON array.
+      JSON.parse(string.gsub("}\n{", "},{").prepend("[").concat("]"))
+    end
+
+    private_class_method def self.to_json_stream(user_proc:)
+      proc do |chunk, bytesize|
+        # Clean up response string of chunks.
+        chunk = chunk.gsub("\n\ndata: [DONE]\n\n", "").gsub("\n\ndata:", ",").gsub("data: ", "")
+
+        # Turn it into JSON.
+        chunk = to_json(chunk)
+
+        # Pass an array of JSONified chunk(s) to the user's Proc.
+        user_proc.call([chunk].flatten, bytesize)
+      end
+    end
+
+    private_class_method def self.conn
+      Faraday.new
```

Comment:
I'm not sure if it's specific to the `faraday` version I'm using, but it looks like the `post` calls (probably all calls) don't actually accept `timeout` as an option, so that is not being applied. I was able to keep the timeout setting by doing this instead of passing `timeout` as an argument to post/get etc:
```suggestion
      Faraday.new do |f|
        f.options[:timeout] = OpenAI.configuration.request_timeout
      end
```

Reply:
Nice spot, have fixed this and changed the tests so they'll catch this issue in future.